sklearn/metrics/_regression.py

```diff
@@ -285,35 +285,43 @@ def mean_pinball_loss(
     >>> from sklearn.metrics import mean_pinball_loss
     >>> y_true = [1, 2, 3]
     >>> mean_pinball_loss(y_true, [0, 2, 3], alpha=0.1)
-    np.float64(0.03...)
+    0.03...
     >>> mean_pinball_loss(y_true, [1, 2, 4], alpha=0.1)
-    np.float64(0.3...)
+    0.3...
     >>> mean_pinball_loss(y_true, [0, 2, 3], alpha=0.9)
-    np.float64(0.3...)
+    0.3...
     >>> mean_pinball_loss(y_true, [1, 2, 4], alpha=0.9)
-    np.float64(0.03...)
+    0.03...
     >>> mean_pinball_loss(y_true, y_true, alpha=0.1)
-    np.float64(0.0)
+    0.0
     >>> mean_pinball_loss(y_true, y_true, alpha=0.9)
-    np.float64(0.0)
+    0.0
     """
-    y_type, y_true, y_pred, multioutput = _check_reg_targets(
-        y_true, y_pred, multioutput
+    xp, _ = get_namespace(y_true, y_pred, sample_weight, multioutput)
+    dtype = _find_matching_floating_dtype(y_true, y_pred, sample_weight, xp=xp)
+
+    _, y_true, y_pred, multioutput = _check_reg_targets(
+        y_true, y_pred, multioutput, dtype=dtype, xp=xp
```

Comment:
We should probably change `_check_reg_targets` to move the `_find_matching_floating_dtype` logic inside it and do it by default, no?

Reply:
TL;DR: I will create a new issue to discuss this.

In addition to regression metrics, the `_check_reg_targets` function is also used in `test_regression.py`. Therefore, updating `_check_reg_targets` would require modifying those tests as well.

Alternatively, we could define a new function, `_check_reg_targets_and_dtype`, in an obvious way and use it specifically for the regression metrics. This approach would allow us to leverage the existing tests with minimal changes.

Perhaps it's worth opening a new issue to discuss this further.